### PR TITLE
m_olib.c OK

### PR DESCRIPTION
**Pull request by shawlucas**
I took pretty much all of this from the Majora's Mask Decompilation, including the documentation. Only difference is the names and 0x4000 is changed to 0x3FFF in a few functions I'm not sure why.

Let me know if there's anything that needs to be changed because it's my first time doing a PR here.

### CODE REVIEW

File: include/z64camera.h
Comment (Prakxo) [inline include/z64camera.h:1]: `z64camera.h` -> `m_camera2.h`

File: src/overlays/actors/ovl_Koinobori/ac_koinobori.c [hunk 1]
```diff
@@ -145,7 +145,7 @@ void aKOI_actor_draw(Actor* thisx, Game_Play* game_play) {
         gSPSegment(__polyOpa++, 0x0A, Lib_SegmentedToVirtual(&aKOI_obj_e_koinobori_a_pal));
         gSegments[6] = OS_K0_TO_PHYSICAL(object);
         gSPSegment(__polyOpa++, 0x06, object);
-        CLOSE_POLY_OPA_DISPS();
+        CLOSE_POLY_OPA_DISPS()
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Koinobori/ac_koinobori.c:148]: I think the `;` styling is preferred

Comment (shawlucas) [reply src/overlays/actors/ovl_Koinobori/ac_koinobori.c:148]: This is actually an error I have no idea why this happened im trying to fix the PR right now

Comment (shawlucas) [reply src/overlays/actors/ovl_Koinobori/ac_koinobori.c:148]: It seems like format.py is automatically removing these semicolons

**Review by Prakxo — APPROVED** [review]

File: src/overlays/actors/ovl_Koinobori/ac_koinobori.c [hunk 2]
```diff
@@ -109,16 +109,22 @@ void aKOI_actor_init(Actor* thisx, Game_Play* game_play) {
 }
 
 #define OPEN_POLY_OPA_DISPS()                 \
+    do {                                      \
+    } while (0);                              \
```
Comment (emilybrooks) [inline src/overlays/actors/ovl_Koinobori/ac_koinobori.c:113]: I think what darkeye was saying was only put 1 do while at the end of each macro. The OPEN_DISPS macro has an extra one at the start for matching reasons, but its unnecessary for these ones.

**Review by emilybrooks — COMMENTED** [review]
Nice work! Sorry about the draw macro stuff, I'm not sure why the format script is causing problems.

**Review by emilybrooks — APPROVED** [review]

**Pull request merged**